frontend/packages/console-shared/src/utils/resource-utils.ts

```diff
@@ -345,7 +346,7 @@ const getReplicationControllerAlerts = (rc: K8sResourceKind): OverviewItemAlerts
   }
 };
 
-const getAutoscaledPods = (rc): PodKind[] => {
+const getAutoscaledPods = (rc): ExtPodKind[] => {
```

Comment:
This is existing code, but I dislike that this method returns something claiming to be a pod that's not. The replication controller `spec` and `status` have completely different properties than pod `spec` and `status`. At this point, I'd almost rather admit this is something completely made up and call it `any`. It's not a `K8sResourceKind`.

Can we revisit how we're handling this? It would be better to avoid creating imaginary types. I have a feeling this is going to lead to bugs down the road.

How are these `ExtPodKind` resources used? What scenario are we handling here and below, and how is it displayed in the UI?

Reply:
We use these to show status on Pod Donuts in topology view. Here Autoscaled is used to show `Autoscaled to 0` on pod donut for knative deployments similarly `getIdledStatus` is also used to show Idled status on the pod donuts.

Reply:
What happens in the details pages for these resources? Does this only work in the topology view?

It seems like this logic should be in the pod donut, not here.

Reply:
To be fair @sahil143 did mention originally that it was going to be a larger change to fix the type issue. I suggested that a quick solution might be to extend the type and return the pod phases back to their original. At the time I thought these types were blocking your work on improving the types. Only later did I see you were able to get around it.

If we're ok with the slower turn around time (ie possibly past 4.3 feature freeze) then your suggestion makes sense to move the logic to the donut itself where it can interpret multiple values to create the visualization instead of relying on made up data to fit the poor needs of the donut.

Reply:
@spadgett Yes, These are also being used in the details page for resources as there is pod donut in both admin and topology view.

Reply:
Sorry, I don't mean to harp on this point :) As long as this works in other contexts like the details pages and admin console workloads tab, I'm good with the short term change.